python_ta/debug/AccumulationTable.py

```diff
@@ -0,0 +1,72 @@
+import inspect
+
+import astroid
+import tabulate
+
+
+class AccumulationTable:
+    _num_iterations: int
+    _accumulation_names: list
+    _loop_accumulation_values: dict
+    _loop_var_name: str
+    _loop_variable: list
+
+    def __init__(self, accumulation_names: list):
+        self._accumulation_names = accumulation_names
+        self._num_iterations = 0
+        self._loop_variable = ["N/A"]
+
+    def add_iteration(self, val: list, var) -> None:
+        self._loop_variable.append(var)
+        for i in range(len(self._accumulation_names)):
+            self._loop_accumulation_values[self._accumulation_names[i]].append(val[i])
+        self._num_iterations += 1
+
+    def _convert_table(self) -> dict:
+        table_dict = {
+            "iterations": list(range(self._num_iterations + 1)),
+            "loop variable (" + self._loop_var_name + ")": self._loop_variable,
+        }
+        for key, pair in self._loop_accumulation_values.items():
+            table_dict[key] = pair
+        return table_dict
+
+    def _tabulate_data(self) -> None:
+        print(
+            tabulate.tabulate(
+                self._convert_table(), headers="keys", colalign=("left", "left", "left")
+            )
+        )
+
+    def _loop_string(self, nconverted: str) -> str:
```

Comment:
You should improve this function in two ways:

1. Make use of the *current line number* to determine where in the source code to start
2. Check the *indentation of all non-blank lines* to determine where in the source code to *end* (remember that you only care about the block of code that's *indented within* the `with` statement).